tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala

```
@@ -293,10 +294,10 @@ class DispatcherSpec extends BaseSpec with DetectPlatform {
           .evalOn(customEc)
           .flatMap(runner =>
             Resource.eval(IO(runner.unsafeToFuture(IO.raiseError(new Exception("boom"))))))
-        reported <- Resource.eval(IO.fromFuture(IO(errorReporter.future)))
-      } yield reported
+      } yield errorReporter
 
-      test.use(t => IO(t).mustFailWith[TimeoutException])
+      test.use(t =>
+        IO.fromFuture(IO(t.future)).timeout(1.second).mustFailWith[TimeoutException])
```

In the sad path, if errors ARE reported, this test will fail fast with
```
[error]     x do not treat exception in unsafeRunToFuture as unhandled
[error]      java.lang.Exception: boom doesn't have class 'java.util.concurrent.TimeoutException'. It has class 'java.lang.Exception' (Runners.scala:96)
```

I can also construct a situation[^1] where the test fails like that, but there is _also_ a stacktrace for `java.lang.IllegalStateException: Promise already completed.` This seems _not correct_ but I'm not sure why this is happening. Anyway just a heads up 😅 

[^1]: By trying to get the `unsafeRunAndForget` test passing, but letting this test fail. I kept the update to `unsafeRunAndForget` but switched back to `onError` in `unsafeToFutureCancelable`

---

I have figured this out! It actually wasn't this test that was causing the stack trace, it was the `unsafeRunAndForget` test, and it makes sense (imo). 

By trying to make only this test fail, I ended up in a situation where `unsafeRunAndForget` had a callback to report the failure, and ALSO `unsafeToFutureCancelable` was using `onError` which would _also_ report the failure. So there were two different, and legit, attempts to complete the `errorReporter` promise.

But part of my fix is that `unsafeToFutureCancelable` _shouldn't_ be reporting the error like that, so I think this extra stack trace is okay because it will always accompany a failing test for `unsafeToFuture`